### PR TITLE
Add unit converter

### LINKED ISSUE #197: unit converter
Pls implement Unit converter like area,length,volume,temperature etc 

rr

### PR DESCRIPTION
**Pull request by esensar**
This closes #197

### CODE REVIEW

**Comment by tibbi** [comment]
at changing values, the current selection isnt marked in the radiogroup

**Comment by tibbi** [comment]
add top activity titles, like Length and Area at the top and change X at back arrows

**Comment by tibbi** [comment]
we could write the short codes like "km" at given units too, both at the radio dialog and at the main screen with inserted value and result

**Comment by tibbi** [comment]
use 0 as default values, not 1

**Comment by esensar** [comment]
> we could write the short codes like "km" at given units too, both at the radio dialog and at the main screen with inserted value and result

I initially skipped this because it will require a bunch more strings for translation and I think some of them would even have to be plurals. I just want to confirm that this is alright before I start adding them all.

**Comment by tibbi** [comment]
> > we could write the short codes like "km" at given units too, both at the radio dialog and at the main screen with inserted value and result
> 
> I initially skipped this because it will require a bunch more strings for translation and I think some of them would even have to be plurals. I just want to confirm that this is alright before I start adding them all.

the vast majority of those should be in donottranslate, as they shouldnt be translated.

**Comment by esensar** [comment]
Well at the very least these should be written in other scripts depending on the language (cyrillic, arabic), so they still have to have some kind of translation.

**Comment by tibbi** [comment]
lets differentiate the units somehow, maybe with a small colored bg, like buttons are. Also make sure to align the right side of values so that "0" in "0km" is at the same width as the "0" in "0m"

**Comment by tibbi** [comment]
also add a Unit Converted title at the activity where length, area etc are picked

**Comment by tibbi** [comment]
and move the Length, Area etc rectangles a bit further from the left and right side by increasing the margin

**Comment by tibbi** [comment]
and allow copying both the inserted and calculated values by long pressing, without unit though. So if you long press "5 km", only "5" should be copied.

File: app/src/main/kotlin/com/simplemobiletools/calculator/activities/UnitConverterActivity.kt [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    private val binding by viewBinding(ActivityUnitConverterBinding::inflate)
+    private var vibrateOnButtonPress = true
+    private lateinit var converter: Converter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        isMaterialActivity = true
+        super.onCreate(savedInstanceState)
+        setContentView(binding.root)
+        appLaunched(BuildConfig.APPLICATION_ID)
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calculator/activities/UnitConverterActivity.kt:32]: Do not call this, it can trigger Rate Us dialog

**Comment by tibbi** [comment]
Length, Area etc should have only outside margins increased, not the onew inbetween them. Looks too bad now. Units like km should be a lot smaller than now aswell, so their corner radius should be smaller too. Test the whole app with max system font size too please, there are quite a few issues to be found.

**Comment by tibbi** [comment]
labels like Length, Temperature etc should be centered horizontally too, even if they span across multiple lines.

**Comment by esensar** [comment]
I had to limit these labels to a single line, since multi line approach looked really bad on big text sizes and was cut off. Scaling it with `AutoFitHelper` looked bad as well. I guess these should rarely be long enough to be cut off and it should still be clear what they are in these cases too.

**Comment by tibbi** [comment]
and "C" should clear 1 character only, not reset everything

**Comment by tibbi** [comment]
lets use even smaller height unit backgrounds and use circular ripple at top/down arrows

**Comment by tibbi** [comment]
would be better to also use smaller buttons to make it have less height, and also make it usable in landscape mode too

**Comment by tibbi** [comment]
try scaling the unit field somehow, if you select "millenium", it is really tiny text, even has dots there. It should stretch properly.

**Comment by tibbi** [comment]
long pressing the top area plays a ripple effect on the bottom unit background, try adding big ripples to whole big areas including inserted values and selected units, like at the sample app.

**Comment by tibbi** [comment]
also in light themes the top area isnt highlighted, lets add some background there, maybe by using the very light primary color, or some light shade of grey, whatever looks better

**Comment by tibbi** [comment]
and in landscape mode it shouldnt scroll, try making the top views smaller

File: app/src/main/res/values-ar/strings.xml [hunk 2]
```diff
@@ -13,11 +14,134 @@
...
+    <string name="unit_time_year_leap">Year (Leap)</string>
+    <string name="unit_time_year_gregorian">Year (Gregorian)</string>
+    <string name="unit_time_year_julian">Year (Julian)</string>
+    <string name="unit_time_jiffy">Jiffy</string>
+    <string name="unit_time_moment">Moment</string>
+    <string name="unit_time_fortnight">Fortnight</string>
+    <string name="unit_time_decade">Decade</string>
+    <string name="unit_time_century">Century</string>
+    <string name="unit_time_millennium">Millennium</string>
+    <string name="unit_time_atomic_unit">Atomic Unit</string>
```
Comment (tibbi) [inline app/src/main/res/values-ar/strings.xml:139]: we have way too many units everywhere, try removing the less popular ones please. Use the sample app for inspiration, no need to go this advanced mode.

**Comment by tibbi** [comment]
long pressing the top field looks weird. Whole rectangle is colored, but ripple has rounded corners.

**Comment by tibbi** [comment]
yeah thanks, looks great now :)

**Pull request merged**